### PR TITLE
fix(components): [input-number] display value don't match actual value

### LINKED ISSUE #14438: [Component] [input-number] el-input-number组件显示值与实际值对不上
Bug Type: **`Component`**

## Environment

- Vue Version: `3.3.4`
- Element Plus Version: `2.3.14`
- Browser / OS: `Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/117.0.0.0 Safari/537.36 Edg/117.0.2045.47`
- Build Tool: `Vite`

### Related Component

- `el-input-number`

### Reproduction Link

[Element Plus Playground](https://element-plus.run/#eyJzcmMvQXBwLnZ1ZSI6Ijx0ZW1wbGF0ZT5cbiAgPGVsLWlucHV0LW51bWJlclxuICAgIHYtbW9kZWw9XCJudW1cIlxuICAgIDptaW49XCIxXCJcbiAgICA6bWF4PVwiMTBcIlxuICAgIGNvbnRyb2xzLXBvc2l0aW9uPVwicmlnaHRcIlxuICAgIDpwcmVjaXNpb249XCIxXCJcbiAgLz5cbiAgPGVsLWJ1dHRvbiBAY2xpY2s9XCJudW0gPSA2LjNcIj4xMjM8L2VsLWJ1dHRvbj5cbiAge3tudW19fVxuPC90ZW1wbGF0ZT5cbjxzY3JpcHQgbGFuZz1cInRzXCIgc2V0dXA+XG5pbXBvcnQgeyByZWYgfSBmcm9tICd2dWUnXG5cbmNvbnN0IG51bSA9IHJlZig5LjIpXG5cbjwvc2NyaXB0PlxuIiwiaW1wb3J0LW1hcC5qc29uIjoie1xuICBcImltcG9ydHNcIjoge31cbn0iLCJ0c2NvbmZpZy5qc29uIjoie1xuICBcImNvbXBpbGVyT3B0aW9uc1wiOiB7XG4gICAgXCJhbGxvd0pzXCI6IHRydWUsXG4gICAgXCJjaGVja0pzXCI6IHRydWUsXG4gICAgXCJqc3hcIjogXCJwcmVzZXJ2ZVwiLFxuICAgIFwidGFyZ2V0XCI6IFwiRVNOZXh0XCIsXG4gICAgXCJtb2R1bGVcIjogXCJFU05leHRcIixcbiAgICBcIm1vZHVsZVJlc29sdXRpb25cIjogXCJCdW5kbGVyXCIsXG4gICAgXCJhbGxvd0ltcG9ydGluZ1RzRXh0ZW5zaW9uc1wiOiB0cnVlLFxuICAgIFwidHlwZXNcIjogW1wiZWxlbWVudC1wbHVzL2dsb2JhbC5kLnRzXCJdXG4gIH0sXG4gIFwidnVlQ29tcGlsZXJPcHRpb25zXCI6IHtcbiAgICBcInRhcmdldFwiOiAzLjNcbiAgfVxufVxuIiwiX28iOnt9fQ==)

### Steps to reproduce

先把数字输入框给一个9.3的默认值，然后改成9.2，再改回9.3，这时候点击按钮修改绑定值，显示值与绑定值就对不上了

### What is Expected?

显示值与绑定值一致

### What is actually happening?

显示值与绑定值不一致

### Additional comments

_(empty)_

### PR DESCRIPTION
**Pull request by betavs**
Please make sure these boxes are checked before submitting your PR, thank you!

- [x] Make sure you follow contributing guide [English](https://github.com/element-plus/element-plus/blob/master/.github/CONTRIBUTING.en-US.md) | ([中文](https://github.com/element-plus/element-plus/blob/master/.github/CONTRIBUTING.zh-CN.md) | [Español](https://github.com/element-plus/element-plus/blob/master/.github/CONTRIBUTING.es.md) | [Français](https://github.com/element-plus/element-plus/blob/master/.github/CONTRIBUTING.fr-FR.md)).
- [x] Make sure you are merging your commits to `dev` branch.
- [x] Add some descriptions and refer to relative issues for your PR.

### <samp>🤖 Generated by Copilot at cc71d85</samp>

Simplify and optimize the code for input-number component. Remove redundant logic and improve accessibility by updating `currentValue`, `userInput`, and `aria-valuenow` consistently.

## Related Issue

Fixes #14438

### <samp>🤖 Generated by Copilot at cc71d85</samp>

*  Simplify the condition for updating `currentValue` and `userInput` data properties ([link](https://github.com/element-plus/element-plus/pull/14439/files?diff=unified&w=0#diff-0ae90cff73c6316f25336e807f41c1ecaca3dfbe4210f80273a4310aa89ac5a8L287-R287))
*  Remove unnecessary ternary and nullish coalescing operators for setting `aria-valuenow` attribute of the inner input element ([link](https://github.com/element-plus/element-plus/pull/14439/files?diff=unified&w=0#diff-0ae90cff73c6316f25336e807f41c1ecaca3dfbe4210f80273a4310aa89ac5a8L308-R308), [link](https://github.com/element-plus/element-plus/pull/14439/files?diff=unified&w=0#diff-0ae90cff73c6316f25336e807f41c1ecaca3dfbe4210f80273a4310aa89ac5a8L325-R320))

### CODE REVIEW

File: packages/components/input-number/src/input-number.vue [hunk 1]
```diff
@@ -284,7 +284,7 @@ watch(
   (value) => {
     const userInput = verifyValue(data.userInput)
     const newValue = verifyValue(value, true)
-    if (!isNumber(userInput) && (!userInput || userInput !== newValue)) {
+    if (isNumber(userInput) === false || userInput !== newValue) {
```
Comment (btea) [inline packages/components/input-number/src/input-number.vue:287]:
```suggestion
    if (!isNumber(userInput) || userInput !== newValue) {
```

Comment (betavs) [reply packages/components/input-number/src/input-number.vue:287]: I think this is better to read the code, don't know what you think?

Comment (btea) [reply packages/components/input-number/src/input-number.vue:287]: `isNumber` will only return `true` or `false`. I think it is more concise to use ` logical NOT` directly.

**Review by btea — APPROVED** [review]

**Review by chenxch — APPROVED** [review]

**Comment by kooriookami** [comment]
> /cc @kooriookami I think this PR should be the prefix of #15066, WDYT?

I think #15066 had fixed this issue, can you have a test?

**Comment by kooriookami** [comment]
> > > /cc @kooriookami I think this PR should be the prefix of #15066, WDYT?
> > 
> > 
> > I think #15066 had fixed this issue, can you have a test?
> 
>  [  ](https://private-user-images.githubusercontent.com/35885819/288603530-30290276-81a0-4ce9-8c2c-b44011f24b5f.gif?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTEiLCJleHAiOjE3MDE5MTU0MTYsIm5iZiI6MTcwMTkxNTExNiwicGF0aCI6Ii8zNTg4NTgxOS8yODg2MDM1MzAtMzAyOTAyNzYtODFhMC00Y2U5LThjMmMtYjQ0MDExZjI0YjVmLmdpZj9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFJV05KWUFYNENTVkVINTNBJTJGMjAyMzEyMDclMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjMxMjA3VDAyMTE1NlomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTRjZjQwNjE5MmRlMzkxYWQzNzA3MDJiYTRjNTMxNTJhMTRjYzk2MGI3YzkwMDliZWEzMDE4ODFlZDgzM2UwZWMmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.1iwTwxE2pum2xkqmG5WZXSv-A8oZhVrOKbuOEgsDMFw) [ ](https://private-user-images.githubusercontent.com/35885819/288603530-30290276-81a0-4ce9-8c2c-b44011f24b5f.gif?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTEiLCJleHAiOjE3MDE5MTU0MTYsIm5iZiI6MTcwMTkxNTExNiwicGF0aCI6Ii8zNTg4NTgxOS8yODg2MDM1MzAtMzAyOTAyNzYtODFhMC00Y2U5LThjMmMtYjQ0MDExZjI0YjVmLmdpZj9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFJV05KWUFYNENTVkVINTNBJTJGMjAyMzEyMDclMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjMxMjA3VDAyMTE1NlomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTRjZjQwNjE5MmRlMzkxYWQzNzA3MDJiYTRjNTMxNTJhMTRjYzk2MGI3YzkwMDliZWEzMDE4ODFlZDgzM2UwZWMmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.1iwTwxE2pum2xkqmG5WZXSv-A8oZhVrOKbuOEgsDMFw)
> 
> This pr isn't fix this problem

Ok, I'll take a look.

**Comment by kooriookami** [comment]
@YLMC2017 I fixed it and added a test.

**Comment by YLMC2017** [comment]
> @YLMC2017 I fixed it and added a test.

Okay,it works.

**Comment by kooriookami** [comment]
Sorry, this PR will cause other problems, and I'll close it. Thanks for your contribution.

**Pull request closed without merging**